Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+    
+    public var isFirstPlay: Bool {
+        get {
+            return self.kavaData.isFirstPlay
+        }
+        set(newValue) {
+            self.kavaData.isFirstPlay = newValue
+        }
+    }
+    
+    public var playerEventsToRegister: [PlayerEvent.Type] {
+        return [
+            PlayerEvent.stateChanged,
+            PlayerEvent.loadedMetadata,
+            PlayerEvent.play,
+            PlayerEvent.pause,
+            PlayerEvent.playing,
+            PlayerEvent.seeking,
+            PlayerEvent.sourceSelected,
+            PlayerEvent.ended,
+            PlayerEvent.playbackInfo,
+            PlayerEvent.trackChanged,
+            PlayerEvent.error
+        ]
+    }
+    
+    public func registerEvents() {
+        PKLog.debug("register player events")
+        
+        self.playerEventsToRegister.forEach { event in
+            PKLog.debug("Register event: \(event.self)")
+            
+            switch event {
+            case let e where e.self == PlayerEvent.stateChanged:
+                self.messageBus?.addObserver(self, events: [e.self]) { [weak self] event in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("state changed event: \(event)")
+                    if type(of: event) == PlayerEvent.stateChanged {
+                        strongSelf.handleStateChanged(event: event)
+                    }
+                }
+                
+            case let e where e.self == PlayerEvent.loadedMetadata:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("loadedMetadata event: \(event)")
+                    strongSelf.sendMediaLoaded()
+                })
+                
+            case let e where e.self == PlayerEvent.play:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("play event: \(event)")
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.playRequest)
+                    
+                    if strongSelf.sentPlaybackPoints[KavaEventType.playReached100Percent] == true {
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.replay)
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.pause:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("pause event: \(event)")
+                    strongSelf.kavaData.isPaused = true
+
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.pause)
+                    strongSelf.reportView()
```

Comment:
is this correct even if we haven't reached 10 sec? KAVA expects to get this every 10 sec, need to make sure the service is aware of this(they should as they get|keep the timers they get from us but please make sure)

Reply:
@vadimkononov please answer this

Reply:
as I talked with Eran K., once paused, stop the timer and report what you have till then